### DISCUSSION TITLE
how use Web Worker Loading in vite

### DISCUSSION THREAD

**Discussion by newives** [hrynko/vue-pdf-embed#202] [Q&A] [answered]
how use [Web Worker Loading](https://github.com/hrynko/vue-pdf-embed?tab=readme-ov-file#web-worker-loading) in vite
Can you provide a demo? ❤️

**Comment by hrynko** [comment, marked as answer]
Hi @newives,

In essence, you just need to install proper versions of `vue-pdf-embed` and `pdfjs-dist` as dependencies.

```sh
npm i vue-pdf-embed@2 pdfjs-dist@3.11.174
```

And then import the PdfWorker from `pdfjs-dist` as a URL (see [`?url`](https://vitejs.dev/guide/assets#explicit-url-imports)) and set it to the `GlobalWorkerOptions` imported from `vue-pdf-embed`.

```vue
<script setup>
import { GlobalWorkerOptions } from 'vue-pdf-embed/dist/index.essential.mjs'
import PdfWorker from 'pdfjs-dist/build/pdf.worker.js?url'

import 'vue-pdf-embed/dist/style/index.css'

GlobalWorkerOptions.workerSrc = PdfWorker

const pdfSource = '<PDF_URL>'
</script>

<template>
  <VuePdfEmbed :source="pdfSource" />
</template>
```

**Reply by nocarroll-v8m** [reply]
Hi @hrynko ,

I came across this discussion while looking for a solution to the [Web Worker Loading](https://github.com/hrynko/vue-pdf-embed?tab=readme-ov-file#web-worker-loading) issue

Is there any way to resolve this in the v1 of `vue-pdf-embed` ?

**Reply by hrynko** [reply]
Hi @nocarroll-v8m,

This issue is only fixed in `v2`, so there is no solution in `v1` yet unfortunately
